### PR TITLE
Fix search mobile

### PR DESCRIPTION
**Pull request by sixhobbits**
* Make search work on mobile 
* Make search bar bigger on Desktop
* Update search parameters so that we show a mix of results instead of one article dominating.

e.g. before pygame only showed the juggling example 

image

now it shows a mix of matches from different articles and sections 

image

search on mobile

https://cln.sh/I3C79C

Note that the tweak means we get more false positives (matches for a search where the keyword isn't present), but these are at the bottom of the results and we think this is a good trade-off for now. Will continue tweaking and improving it.

image

### CODE REVIEW

File: static/search.json [hunk 1]
```diff
@@ -641,6 +641,30 @@
     "heading": "Learn More About Nix",
     "slug": "/programming-ide/getting-started-nix#learn-more-about-nix"
   },
+  {
```
Comment (Minivera) [inline static/search.json:644]: Could this generated file changes be moved to its own PR? It's a lot of lines that don't seem related to this work.

File: views/index.ejs [hunk 2]
```diff
@@ -146,5 +146,11 @@
 
     <script src="/prism.js"></script>
     <script src="/scripts.js"></script>
+    <script>
+      function toggleSearch (){
+        let element = document.getElementById("search");
+        element.classList.toggle("search-mobile");
```
Comment (Minivera) [inline views/index.ejs:152]: Why is this added through a script?

**Comment by Minivera** [comment]
Looking good. Added a few comments, I think we could probably remove the search mobile script thingy then it'd be good to go!

**Review by Minivera — COMMENTED** [review]
Looking good. Added a few comments, I think we could probably remove the search mobile script thingy then it'd be good to go!

File: views/index.ejs (hunk 2 above)
Comment (sixhobbits) [reply views/index.ejs:152]: on mobile, we only show the 'search' button and then expand the text input due to limited screen real estate. Or do you mean something else? What would you suggest instead?

https://cln.sh/Z6Eb9q

Comment (Minivera) [reply views/index.ejs:152]: Ha right, it's used directly in the onclick of the search button. Any reason why this wasn't added to `views/header.ejs` instead?

I might also recommend adding a small if to check if the element exists, to prevent any unwanted errors. Something like this:
```
const element = document.querySelector("#search");
if (element) {
  element.classList.toggle("search-mobile");
}
```

Comment (sixhobbits) [reply views/index.ejs]: hey, sorry for delay - we have made changes as you suggested.

**Review by Minivera — APPROVED** [review]
